### ISSUE TITLE
[Debugger] Step through is interpreting the full flow

### ISSUE THREAD

**Issue by VincentBlondeau** [guillep/pharo-core#16351]
**Bug description**
Step-through could be very slow in some cases compared to the normal execution.
It is interpreting the full flow even if it not making any sense.

**To Reproduce**
1. Create a dummy class (Toto for instance)
2. Add those methods:
```
longRunningMethod
	1 to: 4000 do: [ :each | each factorial ]
```
```
someMethod
	1+1.
	self longRunningMethod.
	1+2
```
3. Execute
`Haba2 new someMethod`
it is fast! (3 secs)
4. Debug
`Haba2 new someMethod`
Click:
Over, Into, Over, Through (on longRunningMethod)
Wait a long time

**Expected behavior**
Not waiting that long for the step 4.

**Screenshots**
Fast case:
![Pharo_AoQwvymKVp](https://github.com/pharo-project/pharo/assets/4706329/49976388-2960-4d5e-9a4a-921b9a32ac8f)

Slow case:
![Pharo_B7igMk6Ukq](https://github.com/pharo-project/pharo/assets/4706329/4ee3ed4c-8649-4ec6-9ed7-8f7ccc66aff2)

**Version information:**
 - OS: Win10
 - Version: N/A
 - Pharo Version: 10, 11, 12

**Expected development cost**
It could be lengthy

**Additional context**
Stepover is fast on this method.
There is maybe a lack of optimization in the creation of the additional context for the jump?
GemStone and VisualWorks don't have this issue.

Lifeware related

**Comment by StevenCostiou** [comment]
Yes the step through interprets each bytecode with single context steps, so it is the slowest possible execution.
I will look into it but perhaps the best solution is to remove the step through and replace it specific actions that appear only contextually (such as jump into block).

**Comment by VincentBlondeau** [comment]
Hi Steven,

I don't think removing the step-through is a good solution. You usually want something quite generic, a mix of jump into block and step over, that could be bonded to only 1 action (and with a single keyboard mapping). Having specific actions depending on the context can be quite challenging for users.

It also would not solve the problem. We could have something like below, where when doing step through, it will first execute some slow computation and then the block.
 
```
longRunningMethodAndThenDo: aBlock
	1 to: 4000 do: [ :each | each factorial ].
	aBlock value.
```

**Comment by StevenCostiou** [comment]
I understand but it is debatable, I don't know if we want genericity everywhere.

What the step through does if I remember correctly is single stepping everything until detecting some context property and then stop. That's bad.

The "generic" aspect makes sometimes people use the stepthrough as a  step over, because in the end the result is more or less the same except you stop in blocks. That's bad.

I think we need to answer another question first : why do you need the step through ? what are your scenarios ?
Then we can perhaps simplify it and perhaps fully dissociate over and through.

**Comment by jordanmontt** [comment]
> solution is to remove the step through and replace it specific actions that appear only contextually (such as jump into block).

I actually think this is a good idea. In my case, the only time in which I want to use "Step through" is only when I want to enter to a block

**Comment by VincentBlondeau** [comment]
Last PR: https://github.com/pharo-project/pharo/pull/18095

**Comment by jecisc** [comment]
The FastStepThrough got integrated in Pharo 11 and 14.

It is disabled by default currently. In order to close this issue maybe we need to enable it by default? And if it is better than the previous version in all cases maybe it should become the only version available? 

Steven will probably know better what are the next steps :)
